draft-ietf-quic-version-negotiation.md

```diff
@@ -210,6 +189,54 @@ balancers to ensure that packets for a given connection are routed to the same
 server.
 
 
+# Server Deployments of QUIC {#server-fleet}
+
+While this document mainly discusses a single QUIC server, it is common for
+deployments of QUIC servers to include a fleet of multiple server instances. We
+therefore define the following terms:
+
+Supported Versions:
+
+: This is the set of versions supported by a given individual server instance.
+
+Fully-Deployed Versions:
+
+: This is the set of QUIC versions that is supported by every single QUIC
+server instance in this deployment.
+
+Partially-Deployed Versions:
+
+: This is the set of versions supported by a given individual server instance,
+but which aren't supported by every single QUIC server instance in this
+deployment.
+
+If a deployment only contains a single server instance, then the Supported
+Versions is equal to the Fully-Deployed Versions, and the Partially-Deployed
+Versions is the empty set.
+
+Conversely, if a deployment contains multiple server instances, it is possible
+for software updates to not happen at the exact same time on all server
+instances. In this scenario, some instances will have a different Supported
```

Comment:
I think that you can just say that an instantaneous upgrade is impossible.  A server deployment that includes multiple server instances cannot upgrade all instances instantaneously.  Even if that were possible, there could still be outdated Version Negotiation packets in flight that might result in negotiation failure if downgrade protection were engaged atomically.

This also applies to a single instance deployment.  Though you might avoid the problem by ensuring that your upgrade involves enough downtime to avoid the in-flight VN problem, you might as well just apply the same logic as a multi-server deployment.

Based on that I would remove the "single server instance" paragraph and replace it with a description of the in-flight VN problem.